docs/specs/tsd-001-deprecation.md

```diff
@@ -0,0 +1,181 @@
+# Plugin and Plugin Option Deprecation
+
+## Objective
+
+Specifies the process of deprecating and removing plugins, plugin settings
+including values of those settings or features.
+
+## Keywords
+
+procedure, framework, all plugins
+
+## Overview
+
+Over time the number of plugins, plugin options and plugin features grow and
+some of those plugins or options are either not relevant anymore, have been
+superseded or subsumed by other plugins or options. To be able to remove those,
+this specification defines a process to deprecate plugins, plugin options and
+plugin features including a timeline and minimal time-frames. Additionally, the
+specification defines a framework to annotate deprecations in the code and
+inform users about such deprecations.
+
+## User experience
+
+In the deprecation phase a warning will be shown at Telegraf startup with the
+following content
+
+```text
+Plugin "inputs.logparser" deprecated since version 1.15.0 and will be removed in 2.0.0: use 'inputs.tail' with 'grok' data format instead
+```
+
+Similar warnings will be shown when removing plugin options or option values.
+This provides users with time to replace the deprecated plugin in their
+configuration file.
+
+After the shown release (`v2.0.0` in this case) the warning will be promoted
+to an error preventing Telegraf from starting. The user now has to adapt the
+configuration file to start Telegraf.
+
+## Deprecation Process
+
+After reaching an agreement in the issue, you can start the deprecation process
+by following the steps below.
+
+### File issue
+
+In the filed issue you should outline which plugin, plugin option or feature
+you want to deprecate and *why*! Determine in which version the plugin should
+be removed.
+Consider moving the plugin to the Influx community repository as an external
+plugin if there is no straight-forward replacement.
```

Comment:
Would we actually deprecate a plugin and remove it to an external plugin if there was no replacement? Later on we talk about not deprecating anything unless there is actually a replacement ;) so these seems to conflict.

my suggestion is to remove this paragraph.